ci/docker-publish.yml

```diff
@@ -55,9 +68,21 @@ jobs:
       # Build and push Docker image with Buildx (don't push on PR)
       # https://github.com/docker/build-push-action
       - name: Build and push Docker image
+        id: build-and-push
         uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
         with:
           context: .
           push: ${{ github.event_name != 'pull_request' }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+
+      # Sign the resulting Docker image digest except on PR
+      # https://github.com/sigstore/cosign
+      - name: Sign the published Docker image
+        if: github.event_name != 'pull_request'
```

Comment:
Was chatting with @steiza, and he brought up an interesting point here. For private repositories, the transparency log will effectively leak the identity of a private repository. We had the thought that we should add an if to only sign for public repositories. Users could obviously opt-in to signing for private repositories, but this would not leak their private repo information by default. 

@mattmoor what do you think?

Reply:
That works for me if there's a way to express that in actions (I think the best UX here would be for it to show up as skipped like it would for PRs).

I suspect the next best would be for us to add an option to cosign that lets us toggle this off (perhaps with a clear warning) when the image is private (I can discuss with @dlorenc @dekkagaijin et al and try to land something in 1.4).

WDYT?

Reply:
I staged a fix in cosign for this here: https://github.com/sigstore/cosign/pull/1116, which would let us drop the `--force` (although I'll probably add a comment indicating that folks comfortable with this can add `--force` to publish to the transparency log).

Reply:
Yep, it's very doable. You could add the step below, and then add the (!(env.IS_PRIVATE)) to your existing if statement and I think it would do the job. 

```yaml

      - name: Confirm this is a public repository
        env: 
          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
        run: |
          TEMP=$(gh api repos/$GITHUB_REPOSITORY --jq '.private')
          echo "IS_PRIVATE=$TEMP" >> $GITHUB_ENV
      - name: Run if public
        if: (!(env.IS_PRIVATE))

```

Reply:
Awesome TIL!  I'll try that out and then update this.  I realize that this'll end up skipping signing entirely, so we may still want to adopt the cosign change and drop this once we cut 1.4.

Reply:
ok, I was able to get this working in my little sample repo:
```yaml
      - name: Confirm this is a public repository
        env:
          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
        run: |
          TEMP=$(gh api repos/$GITHUB_REPOSITORY --jq '.private')
          echo "IS_PRIVATE=$TEMP" >> $GITHUB_ENV

      - name: Sign the container image (if public)
        if: ${{ env.IS_PRIVATE == 'false' }}
        env:
          COSIGN_EXPERIMENTAL: "true"
        run: cosign sign ghcr.io/${{ github.repository }}@${{ steps.push-step.outputs.digest }}
```

This ran fine (with signing) in the public repo, and from a private repo instantiated from it the signing step was skipped.  I'll update the PR with this shortly, thanks for the tip!

Reply:
Ok, cool this should be updated.  When we cut cosign `1.4` we can simplify things, since my PR upstream landed.

Reply:
Given you have the whole event payload to work with already on disk you don't need to make the api call.  Each event has a repository object and that will tell you if it is public or private.

Reply:
Looks like what @steiza also said.  This should be fixed in the latest revision!